lib/swaps/SwapRepository.ts

```diff
@@ -0,0 +1,29 @@
+import { db } from '../types';
+import Bluebird from 'bluebird';
+import { Models } from '../db/DB';
+
+class SwapRepository {
+
+  constructor(private models: Models) {}
+
+  public getSwaps = (): Bluebird<db.SwapDealInstance[]> => {
```

Comment:
I'd call this `getSwapDeals` to be consistent with naming.

Reply:
Better yet, since all we really need are the hashes for now, we'd be reading way more data than needed by pulling back everything. It would be better to query just for the r hash values and call this something like `getRHashes`.